R/correlogram.R

```diff
@@ -0,0 +1,196 @@
+#' inputted as continuous, binary, or ordinal variables.
+#' Note: Binary variables must have a value of 0 or 1.
+#' Note: Ordinal variables must be formatted as factors.
+#' @param fig_colors Allows the user to change the colors of the figure
+#' (defaults are provided). Must be vector of length 3, with color corresponding
+#' to strength of correlation.
+#' @param diagonal Allows user to choose to view the correlogram with diagonal
+#' entries (default is FALSE).
+#'
+#' @return A correlogram.
+#' @export
+#' @import ltm
+#' @import tibble
+#' @import rcompanion
+#' @import cowplot
+#' @import ggcorrplot
+#' @import stringr
+#' @import faux
+#'
+#' @examples
+#' create_correlogram(corr)
+create_correlogram <- function(df,
+                               diagonal = FALSE,
+                               fig_colors = colfun()$fig10_colors) {
+  classes <- numeric()
+  shortcs <- numeric()
+
+  df <- as.data.frame(df)
+
+  if (ncol(df) <= 1) {
+    error_message <- "You must have more than one variable in your
+                              dataframe."
+    stop(error_message)
+  }
+
+  if (any(is.na(df))) {
+    miss_vars <- colnames(df)[colSums(is.na(df) > 0)]
+    warning(paste(
+      "you have a missing value in row(s)",
+      which(rowSums(is.na(df)) > 0), "and column(s)",
+      which(colSums(is.na(df)) > 0)
+    ))
+    df[miss_vars] <- lapply(df[miss_vars], function(x) {
+      ifelse(is.na(x), NA, x)
+    })
+  }
+
+  for (i in seq_along(df)) {
+    column_dat <- df[[i]][!is.na(df[[i]])]
+    ifelse(
+      all(column_dat %in% c(0, 1)),
+      c(classes[i] <- "binary", shortcs[i] <- "b"),
+      ifelse(
+        all(is.numeric(column_dat)),
+        c(classes[i] <- "continuous", shortcs[i] <- "c"),
+        ifelse(all(is.factor(column_dat)), c(
+          classes[i] <- "ordinal",
+          shortcs[i] <- "o"
+        ),
+        stop("Please review your dataframe inputs to ensure correct
+                formatting.")
+        )
+      )
+    )
+  }
+
+  df_attribs <- data.frame(
+    names = c(colnames(df)),
+    category = c(classes),
+    shortc = c(shortcs)
+  )
+
+  mat <- data.frame(matrix(NA, nrow = ncol(df), ncol = ncol(df)))
+  dimnames(mat) <- list(names(df), names(df))
+
+  for (i in seq(1, ncol(df))) {
+    for (j in seq(1, ncol(df))) {
+      xattr <-
+        df_attribs[df_attribs$names %in% names(df)[i], ][["shortc"]]
+      yattr <-
+        df_attribs[df_attribs$names %in% names(df)[j], ][["shortc"]]
+
+      type <- paste0(xattr, yattr)
+      # correlation calculations
+      ifelse(
+        type == "cc",
+        # calculates Pearson correlation with two continuous variables
+        mat[i, j] <- cor(df[, i], df[, j]),
+        ifelse(
+          type %in% c("bb", "cb"),
+          # calculates point biserial correlation with either two binary or a
+          # continuous variable as the x attribute followed by a binary variable
+          # as the y attribute.
```

Comment:
Hi Chen! I implemented @details into the function and corrected the correlation between two binary variables to be the Cramer's V phi association, which should be the correct method. Thank you for the comment, let me know if you see anything else!

Reply:
Hi Abigail, thank you for the updates! I think both Phi Coefficient and Cramer's V Association can measure the association between two binary variables. However, Phi Coefficient ranges from -1 from 1 with a negative value indicating negative correlation and a positive value indicating a positive correlation. In contrast, Cramer's V Association ranges from 0 to 1, which does not imply whether an association is positive or negative. I think Phi Coefficient would be more appropriate in our case, as our Correlogram differentiates positive and negative correlations using red and blue colors. 

Here is a reference article to the Phi Coeffcient approach: https://www.statology.org/phi-coefficient-in-r/.

Reply:
Hi Chen, I ran into a few errors using the phi coefficient approach since it can only utilize a 2x2 matrix, where the Cramer's V association can expand the calculations to include more data. I can try to consolidate the binary variables into a 2x2 matrix and use the phi coefficient instead, it just might take more data manipulation.

Reply:
I was able to update the function to calculate the Phi Coefficient! Let me know if you see anything else that should be updated.